src/EnsembleKalmanProcess.jl

```diff
@@ -163,17 +167,26 @@ function get_N_iterations(ekp::EnsembleKalmanProcess)
 end
 
 """
-    construct_initial_ensemble(prior::ParameterDistribution, N_ens::IT; rng_seed=42) where {IT<:Int}
+    construct_initial_ensemble(prior::ParameterDistribution, N_ens::IT; rng_seed::IT = 42, rng::Union{Random.AbstractRNG,Nothing} = nothing) where {IT<:Int}
 
 Construct the initial parameters, by sampling `N_ens` samples from specified
 prior distribution. Returned with parameters as columns.
 """
-function construct_initial_ensemble(prior::ParameterDistribution, N_ens::IT; rng_seed = 42) where {IT <: Int}
+function construct_initial_ensemble(
+    prior::ParameterDistribution,
+    N_ens::IT;
+    rng_seed::IT = 42,
+    rng::Union{Random.AbstractRNG, Nothing} = nothing,
+) where {IT <: Int}
     # Ensuring reproducibility of the sampled parameter values
-    Random.seed!(rng_seed)
-    parameters = sample_distribution(prior, N_ens) #of size [dim(param space) N_ens]
-    return parameters
+    if rng === nothing
+        rng = Random.seed!(rng_seed)
+    end
+    # on the other hand, if we did pass an explicit rng, we seeded it already
+    return sample_distribution(rng, prior, N_ens) #of size [dim(param space) N_ens]
 end
+construct_initial_ensemble(rng::AbstractRNG, prior::ParameterDistribution, N_ens::IT) where {IT <: Int} =
```

Comment:
Sometimes you use `AbstractRNG`, and others `Random.AbstractRNG`. Can you homogenize to `AbstractRNG`?

Reply:
Done in c1b164d